internal/provider/resource_vault_cluster.go

```diff
@@ -499,7 +528,7 @@ func resourceVaultClusterRead(ctx context.Context, d *schema.ResourceData, meta
 	}
 
 	// The Vault cluster was already deleted, remove from state.
-	if *cluster.State == vaultmodels.HashicorpCloudVault20201125ClusterStateDELETED {
+	if *cluster.State == vaultmodels.HashicorpCloudVault20201125ClusterStateDELETING {
```

Comment:
nit: DELETED

Reply:
I think we should remove this check as DELETE can never happen :( somehow proto field was removed here.